de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/editor/annotations/QueueAnnotationStore.java

```diff
@@ -0,0 +1,156 @@
+package de.fu_berlin.inf.dpp.intellij.editor.annotations;
+
+import de.fu_berlin.inf.dpp.filesystem.IFile;
+import de.fu_berlin.inf.dpp.session.User;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+
+/**
+ * A class to store a limited number of annotations of a certain type. The store
+ * operates like a queue, meaning, if the store is full, the oldest annotation
+ * is removed. This is not done automatically but rather should be done by the
+ * caller by calling {@link #removeIfFull()} before adding new annotations.
+ *
+ * @param <E> the stored annotation type
+ */
+class QueueAnnotationStore<E extends AbstractEditorAnnotation>
```

Comment:
The name "QueueAnnotationStore" sounds as if it a store that contains annotations which are related to a queue, but this is a queue storing annotations. Probably something like AnnotationQueue is better?